src/webgl/primitives.js

```diff
@@ -632,9 +630,103 @@ p5.prototype.torus = function(){
+// in the core/2d_primitives file. Arguments used here may
+// be validated or altered in the 2d_primitives file before
+// they are passed here.
+/////////////////////////
+
+
+p5.RendererGL.prototype.arc = function(){
+  var x       = arguments[0];
+  var y       = arguments[1];
+  var width   = arguments[2];
+  var height  = arguments[3];
+  var start   = arguments[4];
+  var stop    = arguments[5];
+  var mode    = arguments[6];
+  var detailX = arguments[7] || 24;
+  var detailY = 1;
+  var epsilon = 0.00001; // Smallest visible angle on displays up to 4K.
+
+
+  var shape =    Math.abs(start%(Math.PI*2) - stop%(Math.PI*2)) < epsilon ||
+                 mode === 'ellipse' ? 'ellipse' : 'arc';
+
+  var gId = shape+'|'+x+'|'+y+'|'+width+'|'+ height +'|'+start+'|'+
+            stop+'|'+ mode+'|'+ detailX;
+  if(!this.geometryInHash(gId)){
+    var _arc = function(){
+      this.strokeIndices = [];
+      var aAdj = shape === 'ellipse' ? 1 : 0;
+
+      //Define the arc
+      for (var i = 0; i < detailX; i++) {
+        var a = i / (detailX -1 + aAdj);
+        var theta = (stop - start) * a + start;
+        var _x = x + width  * 0.5 * Math.cos(theta);
+        var _y = y + height * 0.5 * Math.sin(theta);
+        var _u = 0.5 + 0.5 * Math.cos(theta);
+        var _v = 0.5 + 0.5 * Math.sin(theta);
+
+        this.vertices.push(new p5.Vector(_x, _y, 0));
+        this.uvs.push([_u,_v]);
+      }
+
+      //Define the center point (Done after the ark because of the else branch)
+      if(mode === 'pie' || mode === 'fan' || Math.abs(stop - start) >= Math.PI){
+        this.vertices.unshift(new p5.Vector(x, y, 0));
+        this.uvs.unshift([0.5,0.5]);
+      } else {
+        var startVert  = this.vertices[0];
+        var endVert    = this.vertices[this.vertices.length -1];
+        var adjCenterX = (startVert.x + endVert.x) / 2;
+        var adjCenterY = (startVert.y + endVert.y) / 2;
+
+        this.vertices.unshift(new p5.Vector(adjCenterX, adjCenterY, 0));
+        this.uvs.unshift([0.5 + (adjCenterX - x) / width,
+                          0.5 + (adjCenterY - y) / height]);
+      }
+
+      //Create the faces and stroke along the edges
+      for (var j = 0; j < this.vertices.length -2; j++) {
+        this.faces.push([0, j+1, j+2]);
+        this.strokeIndices.push([j+1, j+2]);
+      }
+
+      //Finalize curve shape
+      if(shape === 'ellipse'){
+        this.faces.push([0, this.vertices.length - 1, 1]);
+        this.strokeIndices.push([this.vertices.length - 1, 1]);
+      } else if (mode === 'open'){
+        this.faces.push([0, this.vertices.length - 1, 1]);
+      } else if (mode === 'chord'){
+        this.faces.push([0, this.vertices.length - 1, 1]);
+        this.strokeIndices.push([this.vertices.length - 1, 1]);
+      } else if (mode === 'pie'){
+        this.strokeIndices.push([this.vertices.length - 1, 0]);
+        this.strokeIndices.push([0, 1]);
+      } else {console.log('Error, unknown mode');}
```

Comment:
I see this error (unknown mode) when I run an example like this:

```
function setup(){
  createCanvas(windowWidth, windowHeight, WEBGL);
}

function draw(){
  background(220);
  arc(width/4, -height/4, 50, 50, PI/3, 2 * PI/3);
}
```
Probably should be a default type for arc?

Reply:
I made 'OPEN' the default.